### PR TITLE
Added option to set specific revision when using Subversion as SCM

### PR DESCRIPTION
**Pull request by marcovtwout**

### CODE REVIEW

**Comment by marcovtwout** [comment]
Deploy config example:

```
set :svn_revision, 24
```

**Comment by leehambley** [comment]
All the tests failed, could you please try again?

**Comment by marcovtwout** [comment]
Sorry for the many commits, I haven't had time yet to set up local testing.

**Comment by marcovtwout** [comment]
Existing test cases now pass.

I naively tried to add a new test case by copying and modifying it in the same file, but I guess I am missing something. Could you point me in the right direction? https://github.com/marcovtwout/capistrano/commit/14a0a861fb02126137f79f76835ba9b9485c8c13

File: spec/lib/capistrano/svn_spec.rb [hunk 1]
```diff
@@ -59,6 +61,18 @@ module Capistrano
...
+        subject.update
+      end
+    end
+    
+    describe "#update_specific_revision" do
+      it "should run svn update and update to a specific revision" do
+        context.expects(:execute).with(:svn, :update, '--username someuser', '--password somepassword', '--revision 12345')
+        context.expects(:fetch).twice.with(:svn_username).returns('someuser')
+        context.expects(:fetch).twice.with(:svn_password).returns('somepassword')
+        context.expects(:fetch).once.with(:svn_revision).returns('12345')
```
Comment (mattbrictson) [inline spec/lib/capistrano/svn_spec.rb]: I think this should be `context.expects(:fetch).twice`.

… based on the [Travis build](https://travis-ci.org/capistrano/capistrano/builds/99255906) which is showing the failing test:

```
Failures:
  1) Capistrano::Svn::DefaultStrategy#update_specific_revision should run svn update and update to a specific revision
     Failure/Error: subject.update
     Mocha::ExpectationError:
       unexpected invocation: #<#<Class:0x00000003740b78>:0x3740b28>.fetch(:svn_revision)
       unsatisfied expectations:
       - expected exactly once, invoked twice: #<#<Class:0x00000003740b78>:0x3740b28>.fetch(:svn_revision)
       - expected exactly once, not yet invoked: #<#<Class:0x00000003740b78>:0x3740b28>.execute(:svn, :update, '--username someuser', '--password somepassword', '--revision 12345')
       satisfied expectations:
       - expected exactly twice, invoked twice: #<#<Class:0x00000003740b78>:0x3740b28>.fetch(:svn_password)
       - expected exactly twice, invoked twice: #<#<Class:0x00000003740b78>:0x3740b28>.fetch(:svn_username)
     # ./lib/capistrano/scm.rb:51:in `fetch'
     # ./lib/capistrano/svn.rb:12:in `svn'
     # ./lib/capistrano/svn.rb:32:in `update'
     # ./spec/lib/capistrano/svn_spec.rb:77:in `block (3 levels) in <module:Capistrano>'
```

**Comment by marcovtwout** [comment]
Anything left to do before this can be merged?

**Comment by leehambley** [comment]
I'll double check. Thanks for the ping.

**Comment by mattbrictson** [comment]
@marcovtwout Can you add a CHANGELOG entry for this and then squash all the commits down to a single commit? Then I can merge it. Thanks

**Comment by marcovtwout** [comment]
https://github.com/capistrano/capistrano/pull/1573

**Pull request closed without merging**